<launch>
  <arg name="standalone" default="true"/> <!-- debug nodes by setting then standalone -->
  <arg name="MANAGER" value="detic_detection_manager" unless="$(arg standalone)"/>
  <arg name="MANAGER" value="" if="$(arg standalone)"/>
  <arg name="LOAD_STATEMENT" value="load" unless="$(arg standalone)"/>
  <arg name="LOAD_STATEMENT" value="standalone" if="$(arg standalone)"/>

  <arg name="namespace" default="local" />
  <arg name="rviz" default="true" />

  <arg name="input_image" default="/kinect_head/rgb/image_rect_color"/>
  <arg name="input_depth" default="/kinect_head/depth_registered/image"/>
  <arg name="input_camera_info" default="/kinect_head/depth_registered/camera_info"/>
  <arg name="confidence_threshold" default="0.5"/>
  <arg name="out_debug_img"        default="true" />
  <arg name="out_debug_segimg"     default="false" />
  <arg name="model_type"           default="swin" />
  <arg name="vocabulary"           default="custom" />
  <arg name="custom_vocabulary"    default="refrigerator" />

  <arg name="_input_image" default="/$(arg namespace)/decompressed_image"/>
  <arg name="_input_depth" default="/$(arg namespace)/decompressed_depth"/>

  <include file="$(find detic_ros)/launch/sample.launch">
    <arg name="input_image"       value="$(arg input_image)" />
    <arg name="out_debug_img"     value="$(arg out_debug_img)" />
    <arg name="out_debug_segimg"  value="$(arg out_debug_segimg)" />
    <arg name="model_type"        value="$(arg model_type)" />
    <arg name="vocabulary"        value="$(arg vocabulary)" />
    <arg name="custom_vocabulary" value="$(arg custom_vocabulary)" />
    <arg name="namespace" value="$(arg namespace)" />
  </include>

  <group ns='$(arg namespace)'>

    <node name="$(arg MANAGER)" pkg="nodelet" type="nodelet" args="manager"/>

    <include file="$(find fridge_demo2025)/launch/decompress_depth.launch">
      <arg name="input_image" value="$(arg input_image)"/>
      <arg name="input_depth" value="$(arg input_depth)"/>
      <rosparam>
        queue_size: 100
      </rosparam>
    </include>

    <node pkg="nodelet" type="nodelet" name="decompress_points"
          args="$(arg LOAD_STATEMENT) depth_image_proc/point_cloud_xyzrgb $(arg MANAGER)">
      <remap from="rgb/camera_info" to="$(arg input_camera_info)"/>
      <remap from="rgb/image_rect_color" to="$(arg _input_image)"/>
      <remap from="depth_registered/image_rect" to="$(arg _input_depth)"/>
      <rosparam>
        queue_size: 100
      </rosparam>
    </node>

    <node name="tf_transform"
          pkg="nodelet" type="nodelet"
          args="$(arg LOAD_STATEMENT) jsk_pcl_utils/TfTransformCloud $(arg MANAGER)">
      <remap from="~input" to="depth_registered/points"/>
      <rosparam>
        target_frame_id: base_footprint
      </rosparam>
    </node>

  </group>

</launch>
